### ISSUE TITLE
Handling ParamTools error messages

### ISSUE THREAD

**Issue by jdebacker** [PSLmodels/Tax-Calculator#2622]
@hdoupe From the errors reported in Issue #2621, it looks like there has been change in the `paramtools` errors from being strings to dictionaries and therefore an error is occurring on line 282 of `taxcalcio.py` and users don't see the intended validation error message.  No current `taxcalc` unit tests seem to catch this.  Can  you confirm that that this is the case and perhaps propose a fix (and ideally a test to catch this type of thing going forward)?

**Comment by hdoupe** [comment]
Thanks @jdebacker I'll check this out. Nothing has changed in ParamTools but maybe there is something in a dependency.

**Comment by martinholmer** [comment]
Recently @jdebacker asked @hdoupe this question:

> From the errors reported in Issue #2621, it looks like there has been
> change in the paramtools errors from being strings to dictionaries and
> therefore an error is occurring on line 282 of taxcalcio.py and users
> don't see the intended validation error message.  No current taxcalc
> unit tests seem to catch this.  Can you confirm that that this is the
> case and perhaps propose a fix (and ideally a test to catch this type
> of thing going forward)?

And a few days later @hdoupe responded with this comment:

> Thanks @jdebacker I'll check this out.  Nothing has changed in
> ParamTools but maybe there is something in a dependency.

The premise of this response seems to be that the Tax-Calculator crash
reported in issue #2621 is caused by some recent change in a
dependency of ParamTools.  But this premise appears to rule out the
possibility that this bug has been present in Tax-Calculator since
version 3.0.0, which is the version in which the old Parameters class
was replaced by a new Parameters class based on ParamTools.

The following simple experiment with older versions of the taxcalc
package show that the bug recently reported in issue #2621 was not
present in taxcalc version 2.9.0, but first appeared in taxcalc
version 3.0.0.  So, the observation by @jdebacker that the unit test
suite is not strong enough to identify this bug is definitely true.
In fact, the results below show that the transition from taxcalc 2.9
to 3.0 was not tested thoroughly enough to catch this bug.

Here is the reform file I used in my experiment:

<pre>
(py37) policy% cat reformX.json 
{
    "AMEDT_rt": {"2021": 1.8}
}
</pre>

This reform file contains a common oversight when users forget the
rate parameters are decimal fractions rather then percents.  So, in
this case, the user wants to double the additional Medicare tax rate
from 0.9% to 1.8% beginning in 2021.

I installed the pre-ParamTools version of the taxcalc package as
follows:

<pre>
(py37) policy% conda install -c pslmodels taxcalc=2.9.0
</pre>

And I ran the CLI version of the model as follows:

<pre>
(py37) policy% tc --version
Tax-Calculator 2.9.0

(py37) policy% tc cps.csv 2022 --reform reformX.json   
ERROR: 2021 AMEDT_rt value 1.8 > max value 1
ERROR: 2022 AMEDT_rt value 1.8 > max value 1
ERROR: 2023 AMEDT_rt value 1.8 > max value 1
ERROR: 2024 AMEDT_rt value 1.8 > max value 1
ERROR: 2025 AMEDT_rt value 1.8 > max value 1
ERROR: 2026 AMEDT_rt value 1.8 > max value 1
ERROR: 2027 AMEDT_rt value 1.8 > max value 1
ERROR: 2028 AMEDT_rt value 1.8 > max value 1
ERROR: 2029 AMEDT_rt value 1.8 > max value 1
ERROR: 2030 AMEDT_rt value 1.8 > max value 1
USAGE: tc --help
</pre>

The model informed the user that 1.8 is greater than the maximum valid
value of 1.

Now I repeat this experiment using Tax-Calculator version 3.0.0:

<pre>
(py37) policy% conda uninstall taxcalc

(py37) policy% conda install -c pslmodels -c conda-forge taxcalc=3.0.0

(py37) policy% tc --version
Tax-Calculator 3.0.0

(py37) policy% tc cps.csv 2022 --reform reformX.json 
Traceback (most recent call last):
  File "/Users/mrh/anaconda3/envs/py37/bin/tc", line 11, in <module>
    sys.exit(cli_tc_main())
  File "/Users/mrh/anaconda3/envs/py37/lib/python3.7/site-packages/taxcalc/cli/tc.py", line 160, in cli_tc_main
    exact_calculations=args.exact)
  File "/Users/mrh/anaconda3/envs/py37/lib/python3.7/site-packages/taxcalc/taxcalcio.py", line 282, in init
    self.errmsg += pol.parameter_errors
TypeError: can only concatenate str (not "dict") to str
</pre>

This is exactly the model output that @JackLandry reported getting in
issue #2621.  This model crash provides the CLI user with no useful
information about why the model run did not work.  This is a serious
bug, which is why @jdebacker asked @hdoupe to fix it and to add a unit
test that would prevent this bug from reappearing.

(Note: I've repeated this experiment getting the same result using
taxcalc versions 3.1.0, 3.2.0, and 3.2.1.)
